client/boot/locale.js

```diff
@@ -45,7 +40,10 @@ export const setupLocale = ( currentUser, reduxStore ) => {
 		const i18nLocaleStringsObject = JSON.parse( window.i18nLocaleStrings );
 
 		reduxStore.dispatch( setLocaleRawData( i18nLocaleStringsObject ) );
-		const languageSlug = get( i18nLocaleStringsObject, [ '', 'localeSlug' ] );
+
+		// This looks weird, but we really _do_ have an object with an empty string as its key
+		// (see: generate_translated_json in WPCOM)
```

Comment:
Is this comment good to stay as-is, in terms of non-open-source code and overall helpfulness to people seeing it for the first time?

Reply:
I feel that the comment makes a statement about its author -- that they were surprised to learn about the empty key -- more than about the code itself 🙂 

I would replace it with something like:

> The empty string key `[ '' ]` is a location where metadata about the translation file (name of the locale, plurals definitions) are stored.

But for someone familiar with translation files this is a well-known and boring fact 🙂 We access the `[ '' ]` at plenty other places in Calypso.